Add 'how to end a param' (#242)
openfisca/openfisca-doc: source/coding-the-legislation/40_legislation_evolutions.md

- The `end` day is **inclusive**: it is the last day a variable and its formulas are active (and not the first day it is not active anymore).
- The `end` value is a string of format `YYYY-MM-DD` where `YYYY`, `MM` and `DD` are respectively a year, month and day.
- When defining a date, the month is given **before** the day.

## Ending a parameter at a specific date

> Similarly to variables, parameters are prone to disappear as the legislation evolves.

Reviewer: ```suggestion
Like variables, parameters tend to disappear as legislation evolves.
```

> For instance, let's assume that the `housing_allowance` is only defined from the 1st of January 2010 to the 30th of November 2016.

Reviewer: Is that a parameter or a variable?

Author: It's a[ parameter](https://github.com/openfisca/country-template/blob/master/openfisca_country_template/parameters/benefits/housing_allowance.yaml).

Reviewer: I don't really want an answer here 😉 I was noting that this is not clear in the current phrasing and it should be made clear in the text 🙂

> We can end `housing_allowance` at a specific date by simply entering **null** as a value for that date (`2016-12-01` in this case).

Reviewer: ```suggestion
We can end `housing_allowance` at a specific date by simply assigning `null` as a value for that date (`2016-12-01` in this case).
```

> ```yaml
> description: Housing allowance amount (as a fraction of the rent)

Reviewer: In which file is that?

```suggestion
```yaml
# in parameters/housing_allowance.yaml
description: Housing allowance amount (as a fraction of the rent)
```

Author: I added the file reference. However since I changed the end date to the 31/12/2016 in the the doc, the example and the file are no longer consistent.

Reviewer: > the example and the file are no longer consistent

I'm not sure I understand.

I don't think it is very important that the file matches a real one somewhere; this is more to explain where the `housing_allowance` parameter name comes from 🙂

metadata:
  unit: /1
  reference: https://law.gov.example/housing-allowance-rate
documentation: |
  A fraction of the rent.

>   From the 1st of Dec 2016, the housing allowance no longer exists.

Reviewer: This doesn't seem necessary since this is defined already.

values:

>   # This parameter is only defined from the 1st of Jan 2010 to the 30th of Nov 2016.

Reviewer: ```suggestion
  # This parameter is only defined from the 1st of Jan 2010 to the 1st of Dec 2016.
```

  2010-01-01:
    value: 0.25
  2016-12-01:
    value: null
```

> Calling the parameter for a period starting from the 1st of December 2016, **2017** for instance, will render an error:

Reviewer: ```suggestion
Trying to obtain the parameter for a period starting from the 1st of December 2016, **2017** for instance, will raise an error:
```

> Example:
>```py 
>tax_benefit_system.parameters(2017).benefits.housing_allowance
>```
> Output:
>```py
>ParameterNotFoundError: The parameter 'benefits[housing_allowance]' was not found in the 2017-01-01 tax and benefit system.
>```

> 

Reviewer: Add a subtitle to split the content for increased readability 🙂

> The same thing can be done for a scale by adding a **null** value for the end date across all rates.

Reviewer: ```suggestion
The same thing can be done for a scale by assigning a `null` value for the end date across all rates.
```

> The following example of `social_security_contribution` ends as of the 1st of January 2017.

Reviewer: > Use the same date as the previous example to make it easier to follow.
